src/hitch.c

```diff
@@ -2766,27 +2767,47 @@ drop_privileges(void)
 #endif
 }
 
+/* Returns 0 on match, 1 if differ */
+int
+addrinfocmp(struct addrinfo *a, struct addrinfo *b)
+{
+	AN(a);
+	AN(b);
+
+	if (a->ai_flags    != b->ai_flags    ||
+	    a->ai_family   != b->ai_family   ||
+	    a->ai_socktype != b->ai_socktype ||
+	    a->ai_protocol != b->ai_protocol ||
+	    a->ai_addrlen  != b->ai_addrlen)
+		return 1;
+
+	/* disregard ai_canonname and ai_next */
+	return memcmp(a->ai_addr, b->ai_addr, a->ai_addrlen) == 0 ? 0 : 1;
+}
+
 int
 get_backend_addrinfo() {
-	static struct addrinfo *addrinfo;
+	struct addrinfo *result;
 	struct addrinfo hints;
 
 	memset(&hints, 0, sizeof hints);
 	hints.ai_family = AF_UNSPEC;
 	hints.ai_socktype = SOCK_STREAM;
 	hints.ai_flags = 0;
 	const int gai_err = getaddrinfo(CONFIG->BACK_IP, CONFIG->BACK_PORT,
-	    &hints, &addrinfo);
+	    &hints, &result);
 	if (gai_err != 0) {
 		ERR("{getaddrinfo-backend}: %s\n", gai_strerror(gai_err));
 		exit(1);
 	}
-	if (backaddr == 0 ||
-	    memcmp(addrinfo->ai_addr, backaddr->ai_addr,
-		   sizeof(struct sockaddr_storage)) != 0) {
-		backaddr = addrinfo;
+	if (backaddr == 0 || addrinfocmp(result, backaddr) != 0) {
+		if (backaddr == 0)
```

Comment:
Was this supposed to be `!= NULL` instead?

Reply:
Yes, `if (backaddr != NULL || ...) {`.